### PR TITLE
Replace deb packaging with AppImages.

### LINKED ISSUE #2585: Allow Installation of playtests next to stable releases
Requires for Linux:
- installation into different paths like `/usr/lib/openra-playtest`
- different startup scripts `openra-playtest`
- packages need renaming to `openra-playtest`
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/100181-allow-installation-of-playtests-next-to-stable-releases?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #9201: Move DEB packaging to Ubuntu PPA
They will then run in a clean VM similar to official packages which enhances quality. Also people get a repository for automatic updates. We can also add dependencies like libSDL2 for old LTS versions there.

Interested people in maintaining this can have a look at https://launchpad.net/~openra-developers
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/26306502-move-deb-packaging-to-ubuntu-ppa?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #12257: Please provide an AppImage for OpenRA
Hi,

I've noticed you provide Debian packages for OpenRA builds, but I was wondering if you could provide something that would benefit users of even more Linux distributions: an AppImage (website: https://appimage.org). AppImages are a type of cross-distribution packaging format that need not be installed in order to be run. All they need in order to be run is to be made executable (with `chmod +x`) and executed. 

Thanks for your time,
Brenton

### LINKED ISSUE #13374: Provide server-only .deb package
This requires removing all client specific dependencies from the control file and all shortcuts, mime registrations and wrappers (except openra-server).

### LINKED ISSUE #14894: Feature request: Debian package repository
Hi,

This game is great, but the interesting thing is that while you's only provide Debian packages for Linux (instead of say providing RPM and Arch packages as well, or w/e), I think RPM-based distributions have the advantage. See RPM-based distributions have package repositories (i.e. the one in the openSUSE Build Service), hence RPM-based distributions get automatic updates, but Debian-based distributions don't. Users of Debian-based distributions instead have to manually download the Debian package whenever a new release is made and then install it manually with dpkg. This can get very irritating, so I am here to ask for Debian package repositories for OpenRA. 

To be clear I am not whinging that you's only provide Debian packages, I get that users like me should be grateful you's provide any binaries at all!

Thanks for your time,
Brenton

### PR DESCRIPTION
**Pull request by pchote**
This PR applies https://github.com/OpenRA/OpenRAModSDK/pull/71 to our upstream mods and removes the now-superseded deb packaging.

Fixes #2585.
Fixes #12257.
Closes #9201.
Closes #13374 (run the AppImage with `--server`).
Closes #14894.

The generated AppImages have a couple of differences from the SDK versions:
1.  We now rebuild the icon cache as well as the mime cache on first run to fix the broken launcher icons (we'll want to backport this to the SDK).
2. We generate metadata for version updates, but the `master.openra.net/appimagecheck` end point is currently a placeholder that upgrades all mods to TD.

https://github.com/pchote/OpenRA/releases has test packages for the different release types.

Some notes:
* On a clean system players just need to download the appimages for the mods they want, make them executable, and then run them.
* The images should run on any reasonable Linux distro (>= CentOS 6) but, because they don't package mono, some users will first need to set up http://www.mono-project.com/download/stable/#download-lin.
* System integration for launcher shortcuts and protocol handlers requires the optional `appimaged` daemon (download from https://github.com/AppImage/AppImageKit/releases/tag/continuous, make it executable, then run with the --install argument).
* The current/old(?) way to enable updates is to download the optional `AppImageUpdate` utility from https://github.com/AppImage/AppImageUpdate/releases/tag/continuous, rename/move it to `/usr/local/bin`, and make it executable.  Log out/in so that `appimaged` can detect it, then right click on the launcher icon to see the update option.  The upstream AppImage system integration tools appear to be undergoing a rethink and rewrite so this will probably improve in the future.
* I have made sure that `make install` and `make uninstall` continue to work, so downstream packaging should be minimally affected.
* I have tested the system integration on Gnome (2 & 3) and Unity, but none of the other mainstream WMs.
* The download page and version metadata files will need to be reworked before we can push a release with these changes.

### CODE REVIEW

File: packaging/linux/AppRun.in [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+mono_missing_or_old() {
+  command -v mono >/dev/null 2>&1 || return 0
+  MONO_VERSION=$(mono --version | head -n1 | cut -d' ' -f5)
+  [ "$(make_version "${MONO_VERSION}")" -lt "$(make_version "${MINIMUM_MONO_VERSION}")" ] && return 0
+  return 1
+}
+
+if mono_missing_or_old; then
+  if command -v zenity > /dev/null
+  then
```
Comment (abcdefg30) [inline packaging/linux/AppRun.in]: We moved the `then` on the same line as the `if`statement everywhere else.

File: packaging/linux/buildpackage.sh [hunk 2]
```diff
@@ -1,60 +1,136 @@
...
+			install -m644 "icons/${MOD_ID}_${i}.png" "${APPDIR}/openra-${MOD_ID}.png"
+		fi
+	done
+
+	install -d "${APPDIR}/usr/bin"
+
+	sed "s/{MODID}/${MOD_ID}/g" openra.appimage.in | sed "s/{TAG}/${TAG}/g" | sed "s/{MODNAME}/${DISPLAY_NAME}/g" > openra-mod.temp
+	install -m 0755 openra-mod.temp "${APPDIR}/usr/bin/openra-${MOD_ID}"
+
+	sed "s/{MODID}/${MOD_ID}/g" openra-server.appimage.in  > openra-mod-server.temp
```
Comment (abcdefg30) [inline packaging/linux/buildpackage.sh]: Double whitespace in front of the `>`.

**Review by abcdefg30 — DISMISSED** [review]

**Comment by pchote** [comment]
Fixed and rebased.  Updated test images are available from https://github.com/pchote/OpenRA/releases/tag/pkgtest-20180502 if anyone wants to confirm that everything still works.

**Comment by abcdefg30** [comment]
`View FAQ` doesn't work anymore.

**Comment by pchote** [comment]
I spent a bit of time on that, but couldn't get to the bottom of what exactly in the AppImage environment was breaking `xdg-open`.  The inconsistencies in the crash dialog (question mark icon, FAQ button but no wiki button, differences in text) always low-level bugged me, so I decided that a better solution was to change it to a proper `--error` dialog:

crash

Updated test images: https://github.com/pchote/OpenRA/releases/tag/pkgtest-20180504-2

**Comment by pchote** [comment]
Two new commits reduce duplication and add graphical dialogs on KDE:
screen shot 2018-05-05 at 00 02 40
screen shot 2018-05-05 at 00 03 47


Updated (again) test images: https://github.com/pchote/OpenRA/releases/tag/pkgtest-20180504-3

**Review by abcdefg30 — APPROVED** [review]

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/470c5b862c7cc38d1e3a6e8625a6fada302f2efa...027927eda7520298a78730619edc417f6202b995)

**Comment by TheAssassin** [comment]
> We now rebuild the icon cache as well as the mime cache on first run to fix the broken launcher icons (we'll want to backport this to the SDK).

See https://github.com/AppImage/AppImageKit/issues/653. Feel free to add your two cents to the discussion. The more people push the issue, the more likely it is that a solution will be implemented soon.

**Pull request merged**